File: 2338-minimum-consecutive-cards-to-pick-up/minimum-consecutive-cards-to-pick-up.js
var minimumCardPickup = function(cards) {
    let lookup = new Set()
    let output = Infinity
    let j = 0

    while (j < cards.length) {
        let card = cards[j]

        if (card in lookup) {
            let size = j - lookup[card] + 1
            output = Math.min(output, size)
        }
        
        lookup[card] = j 
        j ++ 
    }    

    return output == Infinity? -1 : output
};

File: 2338-minimum-consecutive-cards-to-pick-up/minimum-consecutive-cards-to-pick-up.py
class Solution:
    def minimumCardPickup(self, cards: List[int]) -> int:
        i = 0
        j = 0
        output = float("inf")
        lookup = set()

        while (j < len(cards)):

            while (cards[j] in lookup):
                output = min(output, (j - i + 1))
                lookup.remove(cards[i])
                i += 1

            lookup.add(cards[j])
            j += 1
        
        return -1 if output == float("inf") else output